src/ObjectStatus/ObjectStatus.js

```diff
@@ -0,0 +1,71 @@
+import classnames from 'classnames';
+import CustomPropTypes from '../utils/CustomPropTypes/CustomPropTypes';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { OBJECT_STATUS_SIZES, OBJECT_STATUS_TYPES } from '../utils/constants';
+import 'fundamental-styles/dist/icon.css';
+import 'fundamental-styles/dist/object-status.css';
+
+/** *Object Status* is a short text that represents the semantic status of an object. It has a semantic color and an optional icon.
+ * Typically, the object status is used in the dynamic page header and as a status attribute of a line item in a table. */
+
+const ObjectStatus = React.forwardRef(({ children, className, glyph, indication, inverted, link, onClick, size, status, ...props }, ref) => {
+    const objectStatusClasses = classnames(
+        'fd-object-status',
+        {
+            [`sap-icon--${glyph}`]: !!glyph,
+            [`fd-object-status--indication-${indication}`]: !!indication,
+            'fd-object-status--inverted': inverted,
+            'fd-object-status--large': size === 'l',
+            [`fd-object-status--${status}`]: !!status,
+            ['fd-object-status--link']: !!link || !!onClick
+        },
+        className
+    );
+    const StatusTag = typeof link !== 'undefined' ? 'a' : 'span';
+
+    return (
+        <StatusTag
+            {...props}
+            className={objectStatusClasses}
+            href={link}
+            onClick={onClick}
+            ref={ref}>
+            {children}
+        </StatusTag>
+    );
+});
+
+
+ObjectStatus.displayName = 'ObjectStatus';
+
+ObjectStatus.propTypes = {
+
```

Comment:
Can we add something like:

```
ariaLabel: (props) => {
        if (
            (!props.children && !props.ariaLabel) ||
            (!props.children && typeof props.ariaLabel !== 'string')
        ) {
            return new Error('ariaLabel is required if there is no text inside the ObjectStatus component and must be a string');
        }
    },
```

So that we have a required aria-label for a11y for icon only status labels?

Reply:
Yes will do it